README.md

```diff
@@ -43,47 +43,85 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
 
 ## Usage
 
-### Follow these steps:
+We use [**MLFlow**](https://mlflow.org/) to log hyper-parameters and training artifacts. Firstly, run MLFlow server:
+```sh
+cd AIAgent
+poetry run mlflow server -h 127.0.0.1 -p 8080 --serve-artifacts
+```
+
+_Illustration of game process:_
+<div align="center">
+<img src="resources/game_process_illustration.png" width="50%" title="Illustration of game process"/>
+</div>
+
+As shown in the picture there are two different loops in the _Main Loop_. To run _Hyper-parameters Tuning Loop_ do steps listed in [this chapter](#hyper-parameters-tuning). The second loop runs symbolic execution with given methods and updates the dataset. To run it do [these steps](#dataset-improvement-with-symbolic-virtual-machines).
+The _Main Loop_ implies alternation of these two variants of running as much as you want. After the _Main Loop_ you can run symbolic execution with the trained model (checkout [this chapter](#guide-symbolic-execution-with-trained-model)).
+
+### Hyper-parameters tuning
```

Comment:
1) Yes, it's better to present some example.
2) I can't understand why it is less complete. Could you write more about it, please?
3) Hm. I think it's better to create separate chapter for dataset generation because we need to do it once, not every time when run hyper-parameters tuning. 

But the main pipeline consists of hyper-parameters tuning and `Dataset improvement with Symbolic Virtual Machines` as presented in the sequence diagram.

Reply:
> 1. Yes, it's better to present some example.

👍 

> 2. I can't understand why it is less complete. Could you write more about it, please?

I was under the impression that hyperparameter tuning happens inside the main pipeline and requires server launch

> 3. Hm. I think it's better to create separate chapter for dataset generation because we need to do it once, not every time when run hyper-parameters tuning.

👍 
 
> But the main pipeline consists of hyper-parameters tuning and `Dataset improvement with Symbolic Virtual Machines` as presented in the sequence diagram.

👍